### PR TITLE
chore(puma): RHICOMPL-3059 add Puma restart/shutdown hooks

### PR DESCRIPTION
**Pull request by marleystipich2**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: config/puma.rb [hunk 1]
```diff
@@ -47,3 +47,11 @@
   require 'prometheus_exporter/instrumentation'
   PrometheusExporter::Instrumentation::Puma.start(type: 'puma')
 end
+
+on_worker_shutdown do
+  puts '-------------------------Puma worker shutdown-------------------------'
```
Comment (vkrizan) [inline config/puma.rb:52]: Is there a better alternative using puma's logger?

Comment (skateman) [reply config/puma.rb:52]: We're not sure about the context here :thinking:

Comment (skateman) [reply config/puma.rb:52]: Also this is just temporary

Comment (marleystipich2) [reply config/puma.rb:52]: Yeah, this is just temporary so it won't be staying but if we wanted it to be permanent I could try to think of something better?

Comment (vkrizan) [reply config/puma.rb:52]: ack

**Pull request merged**